handbook/engineering/web/goals.md

```diff
@@ -4,29 +4,44 @@
-   Our code host integrations, which are an implementation of our extension API, are a huge driver of adoption inside companies and multiply the value of extensions by bringing them into code review workflows, but are difficult to discover and setup.
-2. **Bring the extension platform into shape.**
-   Our extension platform includes the workflow around creating, installing and using extensions, the API exposed to developers and its documentation.
-   To grow adoption of extensions, these need to be solid, but they are currently lacking on multiple dimensions.
-   Some API areas like code insights are still in prototype phase and are still undocumented (this blends into goal (3)).
-   Our extension host is also currently implemented in a way that makes it difficult for us to maintain, evolve to enable more use cases and to onboard new teammates into this area of the codebase.
-   Combining this with writing a few smaller extensions (part of goal (3)) allows us to dogfood the experience and inform us where the platform is lacking.
-3. **Build out compelling use cases with the extensions platform.**
-   This includes writing more extensions ourselves, but also extending the extension API with more capabilities to enable more use cases.
-   We have a long list of ideas and use cases customers shared with us that we can solve and/or enable customers to solve with extensions.
-   Some of these integrate Sourcegraph with more external services to suit the setups of more customers, others provide completely new value no other product provides.
+_To reach our long-term goal, we set the following medium-term goals to guide our short-term iteration plans. These medium-term goals are listed in order of rough priority. This means we preference a sooner-listed goal when making progress on one would conflict with another goal, but it doesn't mean we only work on the first goal until it's done – we balance our iterations in sum so we can make some progress on all of these goals over a quarter. Within each goal, the sub-goals are also listed in (stricter) priority order._ 
+
+1. **Make the Sourcegraph web interfaces more consistent and improve discoverability of Sourcegraph features.**
+   1. Make progress on design debt throughout the web app where the web team owns the design: repo page areas and panels, navigation and headers, general theming, and the blob page (file viewer). 
+   1. Build and update designs to help users discover or understand everything Sourcegraph offers, like new navigation menus and header concepts. 
+
+1. **Make Sourcegraph extensions a core part of Sourcegraph users' experiences.**
+   1. Track anonymized, general usage of extensions to determine which extensions are most successful at adding value for our users to inform our future work.
+   1. Build new API endpoints that are robust and obviously useful (used immediately in an extension), to grow adoption of extensions. 
```

Comment:
We mention adding **new** endpoints here, but generally I think removing things to reduce complexity is often just as - if not more - important than adding things. Same goes for renaming, restructuring, changing types, support more return values etc. Can we include that too, so it doesn't sound like the imperative is to add and only add?
We have this captured in the long term goal with "easy to use" etc but are not translating it into medium term goals here.

Reply:
Sure, I can make this "built, maintain, and update"